src/parse.ros2.test.ts

```diff
@@ -281,6 +281,68 @@ describe("parseMessageDefinition", () => {
     ]);
   });
 
+  it("handles type names for fields", () => {
+    expect(parse(`time time`)).toEqual([
+      {
+        definitions: [
+          {
+            name: "time",
+            type: "time",
+            isArray: false,
+            isComplex: false,
+          },
+        ],
+        name: undefined,
+      },
+    ]);
+
+    expect(parse(`time time_ref`)).toEqual([
+      {
+        definitions: [
+          {
+            name: "time_ref",
+            type: "time",
+            isArray: false,
+            isComplex: false,
+          },
+        ],
+        name: undefined,
+      },
+    ]);
+
+    expect(
+      parse(`
+    true true
+    ============
+    MSG: custom/true
+    bool false
+    `),
+    ).toEqual([
+      {
+        definitions: [
+          {
+            name: "true",
+            type: "custom/true",
```

Comment:
hang on, how did we get custom/true here?  `parse()` doesn't know that this `true true` definition lives in the `custom/` package...

Reply:
> // return if the search is in the type name

This is how it's always been I think? You can write just `Vector3 position`, you don't need to write `geometry_msgs/Vector3 position`

Reply:
Yeah, but resolving this depends on knowing the package name of the top level definition (which normally we don't at this layer since we don't know the name). It should be resolved by whoever holds the whole datatype->definition map.